src/IPC/PipeRx.cc

```diff
@@ -1,22 +1,52 @@
-#include <iostream>
 #include "IPC.h"
 
+#include <iostream>
+#include <fcntl.h>      /* O_RDONLY */
+#include <unistd.h>     /* read() close() unlink() */
+#include <sys/stat.h>   /* mkfifo() */
+#include <errno.h>  
+#include <string.h>
+
+#define FIFO "./fifoChannel"
+
 void PipeRx::receive(void) {
     this->setupPipeRx();
 
-    this->fileSizeRx();
-
     this->pipeRx();
 }
 
 void PipeRx::setupPipeRx(void) {
-    ;
-}
-
-void PipeRx::fileSizeRx(void) {
-    ;
+    mkfifo(FIFO, 0666);
+    this->fd = open(FIFO, O_RDONLY);
+    if (fd < 0) {
+        throw std::runtime_error(strerror(errno));
+    }
+    std::cout << "Rx opened" << std::endl;
 }
 
 void PipeRx::pipeRx(void) {
-    ;
+    ssize_t count;
+
+    while (1) {
+        count = read(this->fd, this->writeBuf, BUFFERSIZE);
+
+        /* end-of-stream */
+        if (count == 0)
+            break;
+        
+        else {
+            if (count < BUFFERSIZE) {
+                this->writeBuf[count] = '\0';  /* necessary to print out buffer partially */
+            }
+            #if PRINT
+            std::cout << this->writeBuf << std::endl;
```

Comment:
even if your count==BUFFERSIZE the last writeBuf element can be not null. Generally printing char array is a security issue.  I understand is only for debugging purposes, but you can just use std::string(writeBuf, sizeof(writeBuf)) inside cout to avoid the problem